### PR TITLE
Okta detections pt1

### PR DESCRIPTION
**Pull request by calkim-panther**
### Background

Okta Detections
* okta_app_refresh_access_token_reuse.py
* okta_app_unauthorized_access_attempt.py
* okta_group_admin_role_assigned.py
* okta_rate_limits.py

### Testing

image
image

### CODE REVIEW

File: global_helpers/panther_base_helpers.py [hunk 1]
```diff
@@ -200,10 +200,17 @@ def box_parse_additional_details(event: dict):
 def okta_alert_context(event: dict):
     """Returns common context for automation of Okta alerts"""
     return {
+        "event_type": event.get("eventType", ""),
```
Comment (edyesed) [inline global_helpers/panther_base_helpers.py:203]: this is spelled `eventType` in the helper and `eventtype` in the detection

File: rules/okta_rules/okta_app_refresh_access_token_reuse.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+def rule(event):
+    return event.get("eventtype") in (
+        "app.oauth2.as.token.detect_reuse",
+        "app.oauth2.token.detect_reuse",
+    )
+
+
+def title(event):
+    return "Okta Access Token Reuse Attempted by " f"[{event.get('client', {}).get('ipAddress')}]"
```
Comment (edyesed) [inline rules/okta_rules/okta_app_refresh_access_token_reuse.py:12]: it'd probably be helpful to include the `actor:displayName` as well here.

File: rules/okta_rules/okta_app_unauthorized_access_attempt.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+from panther_base_helpers import deep_get, okta_alert_context
+
+
+def rule(event):
+    return event.get("eventtype", "") == "app.generic.unauth_app_access_attempt"
+
+
+def title(event):
+    return (
+        f"[{deep_get(event, 'actor', 'alternateId', default = 'id-not-found')}] "
```
Comment (edyesed) [inline rules/okta_rules/okta_app_unauthorized_access_attempt.py:10]:
```suggestion
        f"[{deep_get(event, 'actor', 'alternateId', default = '<id-not-found>')}] "
```

File: rules/okta_rules/okta_app_unauthorized_access_attempt.yml [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        client:
+            device: Computer
+            geographicalContext:
+                city: Springfield
+                country: United States
+                geolocation:
+                    lat: 11.111
+                    lon: -70
+                postalCode: "1234"
+                state: California
```
Comment (edyesed) [inline rules/okta_rules/okta_app_unauthorized_access_attempt.yml:99]: Springfield, CA -> This is now canon

File: rules/okta_rules/okta_group_admin_role_assigned.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+from panther_base_helpers import okta_alert_context
+
+
+def rule(event):
+    return event.get("eventtype", "") == "group.privilege.grant"
+
+
+def title(event):
+    # pylint: disable=W0613
+    return "Okta Admin Privileges Assigned to Group"
```
Comment (edyesed) [inline rules/okta_rules/okta_group_admin_role_assigned.py:10]: I would include an actor who did the assigning and the group to which admin privileges were assigned.

Comment (calkim-panther) [reply rules/okta_rules/okta_group_admin_role_assigned.py:10]: Unfortunately no sample logs to go off of so I'm not sure what fields would be available

File: rules/okta_rules/okta_rate_limits.yml [hunk 6]
```diff
@@ -0,0 +1,233 @@
+AnalysisType: rule
+Description: Potential DoS/Bruteforce attack or hitting limits (system degradation)
+DisplayName: Okta Rate Limits
+Enabled: true
+Filename: okta_rate_limits.py
```
Comment (edyesed) [inline rules/okta_rules/okta_rate_limits.yml:5]: this is mitre technique `T1110`

File: rules/okta_rules/okta_rate_limits.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+def rule(event):
+    eventtype = event.get("eventtype", "")
+    for detection_event in DETECTION_EVENTS:
+        if fnmatch(eventtype, detection_event):
+            return True
+    return False
+
+
+def title(event):
+    return f"Okta Rate Limit Event: [{event.get('eventtype','')}]"
```
Comment (edyesed) [inline rules/okta_rules/okta_rate_limits.py:23]: please include for which actor

File: rules/okta_rules/okta_group_admin_role_assigned.yml [hunk 8]
```diff
@@ -0,0 +1,156 @@
+AnalysisType: rule
+Description: Detect when an admin role is assigned to a group
+DisplayName: Okta Group Admin Role Assigned
+Enabled: true
+Filename: okta_group_admin_role_assigned.py
```
Comment (edyesed) [inline rules/okta_rules/okta_group_admin_role_assigned.yml:5]: I think of this as [T1548](https://attack.mitre.org/techniques/T1548/), though att&ck suggests that is specific to operating systems.

File: rules/okta_rules/okta_app_unauthorized_access_attempt.yml [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+            id: aaa-bbb-123
+            type: WEB
+        uuid: aa-11-22-33-44-bb
+        version: "0"
+      Name: Other Event
+DedupPeriodMinutes: 60
+LogTypes:
+    - Okta.SystemLog
+RuleID: Okta.App.Unauthorized.Access.Attempt
+Threshold: 1
```
Comment (edyesed) [inline rules/okta_rules/okta_app_unauthorized_access_attempt.yml:156]: It may make sense to up the threshold from 1 on this one.

**Review by edyesed — CHANGES_REQUESTED** [review]
The blocking change is the spelling of `eventType` in the global_helper, it is not aligned with the casing of `eventtype` in the test cases and in the detections 

I believe it would be helpful for users if some of the titles were extended to transfer who the actor is and what resource they have affected.

**Review by edyesed — APPROVED** [review]
bundle of okta detections for refresh tokens getting reused, user getting authZ denied to an application, Admin perms getting attached to a group, and okta rate limits getting reached.

**Pull request merged**